semgrep-core/src/core/Metavariable.ml

```diff
@@ -89,18 +95,21 @@ let mvalue_to_any = function
   | Args x -> G.Args x
   | T x -> G.T x
   | P x -> G.P x
+  | Text (s, info) -> G.E (G.L (G.String (s, info)))
 
 (* This is used for metavariable-pattern: where we need to transform the content
  * of a metavariable into a program so we can use evaluate_formula on it *)
-let program_of_mvalue : mvalue -> G.program option = function
+let program_of_mvalue : mvalue -> G.program option =
+ fun mval ->
+  match mval with
   | E expr -> Some [ G.exprstmt expr ]
   | S stmt -> Some [ stmt ]
   | Id (id, Some idinfo) -> Some [ G.exprstmt (G.N (G.Id (id, idinfo))) ]
   | Id (id, None) -> Some [ G.exprstmt (G.N (G.Id (id, G.empty_id_info ()))) ]
   | N x -> Some [ G.exprstmt (G.N x) ]
   | Ss stmts -> Some stmts
-  | Args _ | T _ | P _ ->
-      logger#debug "program_of_mvalue: Args | T | P are not yet handled!";
+  | Args _ | T _ | P _ | Text _ ->
```

Comment:
We can convert `Text` into a program:
```
Text s -> Some [ G.exprstmt (G.L (G.String (s, info))) ]
```
Maybe `program_of_mvalue` could be written in terms of `mvalue_to_any` since there is some duplication.

Reply:
Well in theory we match the raw text inside a string, so it's not a valid program in the host language. It could be a program in another language, in which case we would do it via the metavariable-pattern: language: we talked about.